### PR TITLE
Handshaker plugin mechanism

### PR DESCRIPTION
**Pull request by markdroth**
This provides the mechanism needed to trigger the creation of new handshakers via plugins.

This PR includes build changes, so it can't be merged until the build moratorium is lifted.

### CODE REVIEW

**Comment by markdroth** [comment]
The interop failure is another case of #8989.

**Comment by markdroth** [comment]
And the "basic tests linux" failure looks like another case of #9007.

File: src/core/lib/channel/handshaker_factory.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ */
+
+#include "src/core/lib/channel/handshaker_factory.h"
+
+void grpc_handshaker_factory_add_handshakers(
+    grpc_exec_ctx *exec_ctx, grpc_handshaker_factory *handshaker_factory,
+    const grpc_channel_args *args, grpc_handshake_manager *handshake_mgr) {
+  if (handshaker_factory != NULL) {
+    handshaker_factory->vtable->add_handshakers(exec_ctx, handshaker_factory,
```
Comment (jboeuf) [inline src/core/lib/channel/handshaker_factory.c]: Maybe check for vtable != NULL here and below.

Comment (markdroth) [inline src/core/lib/channel/handshaker_factory.c]: Done.

**Comment by markdroth** [comment]
Commits have now been squashed.

**Comment by markdroth** [comment]
The one test failure here looks to be another instance of #8977.

**Pull request merged**